lms/djangoapps/discussion_api/tests/test_views.py

```diff
@@ -1566,3 +1759,28 @@ def test_pagination(self):
             404,
             {"developer_message": "Page not found (No results on this page)."}
         )
+
+    def test_profile_image_requested_field(self):
+        """
+        Tests all comments retrieved have user profile image details if called in requested_fields
+        """
+        self.register_get_user_response(self.user)
+        cs_comment_child = self.make_comment_data('test_child_comment', self.comment_id, children=[])
+        cs_comment = self.make_comment_data(self.comment_id, None, [cs_comment_child])
+        cs_thread = make_minimal_cs_thread({
+            'id': self.thread_id,
+            'course_id': unicode(self.course.id),
+            'children': [cs_comment],
+        })
+        self.register_get_thread_response(cs_thread)
+        self.register_get_comment_response(cs_comment)
+        self.create_profile_image(self.user, get_profile_image_storage())
+
+        response = self.client.get(self.url, {'requested_fields': 'profile_image'})
+        self.assertEqual(response.status_code, 200)
+        response_comments = json.loads(response.content)['results']
+
+        for response_comment in response_comments:
+            expected_profile_data = self.get_expected_user_profile(response_comment['author'])
+            response_users = response_comment['users']
+            self.assertEqual(expected_profile_data, response_users[response_comment['author']])
```

Comment:
@jcdyer ah! Right; I got it confusing with `views.py`.
For `test_views.py` and also for `test_api.py` (which is even larger more than 3000 line), I believe we need to clean both these files as Ii see a lot of repetitive test code in these two files which I believe could be consolidated at one place.

Also; what is general convention in edx to split tests in multiple files? 
One possibility is what you suggested; to split list and retrieve tests in different files.
Another can be to split feature/view based tests in separate files (e.g. one file could be for threads and another for comments).